docs/tutorials/03_minimum_eigen_optimizer.ipynb

```diff
@@ -293,6 +297,270 @@
+      "[001]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[101]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[011]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[111]: fval: 4.0000, probability: 0.12 [SUCCESS]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print_samples(qaoa_result.samples, max_len=10)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "We may also want to filter samples according to their status or to a certain percentile, i.e. with probability higher\n",
+    "than some threshold."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "outputs": [],
+   "source": [
+    "def get_filtered_samples(samples: List[SolutionSample],\n",
+    "                         percentile: float = 0,\n",
+    "                         allowed_status: Tuple[OptimizationResultStatus] = (OptimizationResultStatus.SUCCESS,)):\n",
+    "    res = []\n",
+    "    for s in samples:\n",
+    "        if s.status in allowed_status and s.probability > percentile:\n",
+    "            res.append(s)\n",
+    "\n",
+    "    return res"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 51,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[010]: fval: -2.0000, probability: 0.12 [SUCCESS]\n",
+      "[000]: fval: 0.0000, probability: 0.12 [SUCCESS]\n",
+      "[110]: fval: 0.0000, probability: 0.12 [SUCCESS]\n",
+      "[100]: fval: 1.0000, probability: 0.12 [SUCCESS]\n",
+      "[001]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[101]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[011]: fval: 3.0000, probability: 0.12 [SUCCESS]\n",
+      "[111]: fval: 4.0000, probability: 0.12 [SUCCESS]\n"
+     ]
+    }
+   ],
+   "source": [
+    "filtered_samples = get_filtered_samples(qaoa_result.samples,\n",
+    "                                        percentile=0.005,\n",
+    "                                        allowed_status=(OptimizationResultStatus.SUCCESS,))\n",
+    "print_samples(filtered_samples, max_len=10)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "If we want to obtain a better perspective of the results, statistics is very helpful, both with respect to\n",
+    "optimized function values and their respective probabilities. Thus, mean and standard deviation are the very basics for\n",
+    "understanding the results."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 52,
+   "outputs": [],
+   "source": [
+    "class PropertyChoice(Enum):\n",
```

Comment:
You don't need to introduce another Enum for filtering.
You can just generate two list
```
fvals = [sample.fval for sample in result.samples]
probs = [sample.probability for sample in result.samples]
```